### ISSUE TITLE
Fast and correct type parsing written by Elm core team

### ISSUE THREAD

**Issue by joonazan** [Bogdanp/elm-ast#81]
I found that the package.elm-lang.org frontend perfectly solves the problem for which I have used this library. The type parsing code there might be helpful in this project.

https://github.com/elm-lang/package.elm-lang.org/blob/master/src/frontend/Parse/Type.elm

**Comment by wende** [comment]
Looks good. Might help in some areas

**Comment by baransu** [comment]
There is unreleased [official Elm parser](https://github.com/elm-lang/parser). It is very simple and has different approach than regexow based parsers but may be worth experimenting with. It's utility library for parses and may be useful in some areas as well

**Comment by brian-carroll** [comment]
@joonazan that code was recently removed, so the link is broken now. Thanks for sharing though. Here's the relevant commit:

https://github.com/elm-lang/package.elm-lang.org/commit/e2bb304e78377514335059850922743c82d33970

**Comment by joonazan** [comment]
@brian-carroll Do you know why package.elm-lang.org does not need it anymore?

Also, if anyone is interested, I can open-source a version of the type parser with the JS translated into pure Elm. I'm using it in a closed-source project of mine.

**Comment by brian-carroll** [comment]
@joonazan no, sorry, I don't know any more. The commit message says. "Superseded by elm-tools/parser and elm-tools/documentation" but I didn't dig any deeper.
